src/test/java/io/gravitee/el/spel/context/SpelTemplateEngineTest.java

```diff
@@ -688,4 +711,100 @@ public void shouldGetValueAsBoolean() {
+        strings = {
+            "/api/[0-9]{2}/(.*)",
+            "/api/[0-9]{217}/(.*)",
+            "{",
+            "{\n  \"status\": \"OK\"\n}",
+            "{\t\"status\": \"OK\"  }",
+            "{ \"status\": \"OK\"  }",
+            "{ \"status\": \"{2}\"  }",
+            "{ \"status\": \"'{2}'\"  }",
+        }
+    )
+    void shouldEvaluateSimpleStringWithoutExpressions(String expression) {
+        final String evaluatedExpression = TemplateEngine.templateEngine().getValue(expression, String.class);
+        assertEquals(expression, evaluatedExpression);
+    }
+
+    private static Stream<Arguments> expressionsWithoutVariables() {
+        return Stream.of(
+            Arguments.of("{2} {('abc'.matches('[0-9]{2}'))} {3}", "{2} false {3}"),
+            Arguments.of("{2} {a} {('78'.matches('[0-9]{2}'))} {3}", "{2} {a} true {3}"),
+            Arguments.of("{2} {('718'.matches('[0-9]{2}'))} {3}", "{2} false {3}"),
+            Arguments.of("{T(java.lang.String).format(\"%scd\", \"ab\")}", "abcd"),
+            Arguments.of("{T(java.lang.String).format(\"%scd\", \"{2}\")}", "{2}cd"),
+            Arguments.of("{  T ( java.lang.String ).format(\"%scd\", \"ab\")}", "abcd"),
+            Arguments.of("{ T ( java.lang.String ).format(\"%scd\", \"{2}\")}", "{2}cd"),
+            Arguments.of("{1 == 1}", "{1 == 1}"),
+            Arguments.of("{(1 == 1)}", "true"),
+            Arguments.of("{(12 == 1)}", "false")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("expressionsWithoutVariables")
+    void shouldEvaluateStringWithExpression(String expression, String expectedResult) {
+        final String evaluatedExpression = TemplateEngine.templateEngine().getValue(expression, String.class);
+        assertEquals(expectedResult, evaluatedExpression);
+    }
+
+    private static Stream<Arguments> expressionsWithVariables() {
+        return Stream.of(
+            Arguments.of("{2} {#request.pathInfo.matches('[0-9]{2}')} {3}", "{2} false {3}"),
+            Arguments.of("{2} {#request.pathInfo.matches('/my/path/[A-Z]{1}[0-9]{2}')} {3}", "{2} true {3}"),
+            Arguments.of("{2} {#request.pathInfo.matches('/my/path/[A-Z]{2}[0-9]{2}')} {3}", "{2} false {3}"),
+            Arguments.of("{2} {  #  request.pathInfo.matches ( '/my/path/[A-Z]{1}[0-9]{2}')} {3}", "{2} true {3}"),
+            Arguments.of("{2} {  #  request.pathInfo.matches ( '/my/path/[A-Z]{2}[0-9]{2}')} {3}", "{2} false {3}"),
+            Arguments.of("{ '/my/path/A58' == #request.pathInfo}", "{ '/my/path/A58' == #request.pathInfo}"),
+            Arguments.of("{('/my/path/A58'==#request.pathInfo)}", "true"),
+            Arguments.of("{('/my/path/A59'==#request.pathInfo)}", "false")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("expressionsWithVariables")
+    void shouldEvaluateStringWithExpressionContainingVariables(String expression, String expectedResult) {
+        lenient().when(request.pathInfo()).thenReturn("/my/path/A58");
```

Comment:
🤔  I had to add this `lenient` other the case:
```java
Arguments.of("{ '/my/path/A58' == #request.pathInfo}", "{ '/my/path/A58' == #request.pathInfo}"),
```

is throwing:

```shell
org.mockito.exceptions.misusing.UnnecessaryStubbingException: 
Unnecessary stubbings detected.
Clean & maintainable test code requires zero unnecessary code.
Following stubbings are unnecessary (click to navigate to relevant line of code):
  1. -> at io.gravitee.el.spel.context.SpelTemplateEngineTest.shouldEvaluateStringWithExpressionContainingVariables(SpelTemplateEngineTest.java:803)
Please remove unnecessary stubbings or use 'lenient' strictness. More info: javadoc for UnnecessaryStubbingException class.
```

Reply:
yes, sounds ok to me :+1: